### ISSUE TITLE
Can I logging assertions(success and fails) or add assertions to steps?

### ISSUE THREAD

**Issue by VictorYakushenko** [fluentassertions/fluentassertions#1411]
### Description

I would like to log assertions into steps (Allure for example) and other logs. Can I just add a logger or another handler so I don't have to do it for every assertion? Like [HttpMessageHandler in HttpClient](https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/httpclient-message-handlers)

**Comment by dennisdoomen** [comment]
No, FA doesn't have any built-in logging, with the exception of the `Should().BeEquivalentTo()` API. That one supports a `ITraceWriter`.

**Comment by VictorYakushenko** [comment]
Are you planning to add a handler for adding some decorators in the future?

**Comment by dennisdoomen** [comment]
I'm not sure what you mean. What kind of logging are you trying to collect?

**Comment by VictorYakushenko** [comment]
For example, wrap assertion in step of report(for example Allure) and add assertion description as step description 
![image](https://user-images.githubusercontent.com/19596916/96589180-30c99b00-12ed-11eb-9dae-c89b023e1ac6.png)

**Comment by dennisdoomen** [comment]
Ah, maybe you can use the `ThrowException` delegate on the `Services` class. See https://github.com/fluentassertions/fluentassertions/blob/344784de5fa25f56dbf2d8a8f8d0787db2a985e6/Src/FluentAssertions/Common/Services.cs#L37

**Comment by VictorYakushenko** [comment]
Can I use it when success assertion? DelegatingHandler in HttpClient is the example that most fits. Or like ThrowException but for the entire assertion:
This is the dummy code below just for the example
`        public static Action<AssertionScope, string> AssertionHandler { get; set; }

        public void SomeHandler(AssertionScope scope, string description)
        {
            Console.WriteLine($"Before Assertion: {description}");
            scope.ExecuteAssertion();
            Console.WriteLine($"After Assertion: {description}");
        }`

**Comment by dennisdoomen** [comment]
> Can I use it when success assertion? DelegatingHandler in HttpClient is the example that most fits. Or like ThrowException but for the entire assertion:

No, I can't think of a way to do that. It wasn't designed for that purpose.

**Comment by VictorYakushenko** [comment]
Ok, Thanks

**Comment by xadvfh** [comment]
This would really be a great thing to have.  Lots of companies are requiring logs/reports for audit purpose and it would be nice to have a way to record pass and failures

**Comment by WolfBublitz** [comment]
Are there any new plans to introduce such a feature?

**Comment by VictorYakushenko** [comment]
Many people need this functionality, are there any options to implement it?

**Comment by jnyrup** [comment]
There no new plans, the design hasn't changed since: 
> It wasn't designed for that purpose.


I'm still not sure what it is that you want to log and where you want to invoke/control this.
Please provide a _complete_ code example of what you imagine.
